.github/workflows/redpanda-build.yml

```diff
@@ -24,11 +24,36 @@ jobs:
           run: |
             export DEBIAN_FRONTEND=noninteractive
             apt-get update -y
-            apt-get install -y git sudo pkg-config
+            apt-get install -y git sudo pkg-config ccache
 
         - name: install dependencies
           run: ./install-dependencies.sh
 
+        # there is no analog to ${{ github.workspace }} for the home directory
+        - name: prepare cache variables
+          id: prepare_cache_variables
+          run: |
+            echo ::set-output name=home::$HOME
+            echo ::set-output name=timestamp::$(date +'%Y-%m-%d-%H-%M-%S')
+
+        - name: ccache cache files
+          uses: actions/cache@v2
+          with:
+            path: ${{ steps.prepare_cache_variables.outputs.home }}/.ccache
+            key: ccache-${{ steps.prepare_cache_variables.outputs.timestamp }}
+            restore-keys: ccache-
```

Comment:
this was inspired by this guide:
https://cristianadam.eu/20200113/speeding-up-c-plus-plus-github-actions-using-ccache/